### PR TITLE
Model#insertContent() should properly auto-paragraph nodes

### LINKED ISSUE #8953: Single paragraph with different text formatting breaks it into multiple paragraphs while pasting on image
## 📝 Provide detailed reproduction steps (if any)

1.  Set different text formatting on parts of a paragraph
2.  Copy content of the paragraph
3.  Paste over the image

### ✔️ Expected result

An image is replaced with a single paragraph that looks exactly like the original one.

## 📃 Other details

*   Browser: …
*   OS: …
*   First affected CKEditor version: very old bug
*   Installed CKEditor plugins: …

---

If you'd like to see this fixed sooner, add a 👍 reaction to this post.

### PR DESCRIPTION
**Pull request by niegowski**
### Suggested merge commit message ([convention](https://ckeditor.com/docs/ckeditor5/latest/framework/guides/contributing/git-commit-message-convention.html))

Fix (engine): Pasting formatted single-line text over a widget should not split it into multiple paragraphs. Closes #8953.

---

### Additional information

### CODE REVIEW

File: packages/ckeditor5-engine/src/model/utils/insertcontent.js [hunk 1]
```diff
@@ -226,6 +239,33 @@ class Insertion {
...
+		// If the real end was after the last auto paragraph then update relevant properties.
+		if ( positionAfterNode.isAfter( positionAfterLastNode ) ) {
+			this._lastNode = node;
+
+			/* istanbul ignore if */
+			if ( this.position.parent != node || !this.position.isAtEnd ) {
+				// Algorithm's correctness check. We should never end up here but it's good to know that we did.
+				// At this point the insertion position should be at the end of the last auto paragraph.
+				// Note: This error is documented in other place in this file.
+				throw new CKEditorError( 'insertcontent-invalid-insertion-position', this );
```
Comment (niegowski) [inline packages/ckeditor5-engine/src/model/utils/insertcontent.js:261]: This is a clone of the previous check that is in `_mergeOnRight()` but this is a slightly different scenario.

**Comment by scofalik** [comment]
LGTM. Do we want @ckeditor/qa-team to check it before merging?

**Comment by niegowski** [comment]
> Do we want @ckeditor/qa-team to check it before merging?

I guess we should 😉

**Comment by FilipTokarski** [comment]
LGTM 👍

**Pull request merged**